satpy/readers/iasi_l2_so2_bufr.py

```diff
@@ -0,0 +1,238 @@
+-------
+Here is an example how to read the data in satpy:
+
+.. code-block:: python
+
+    from satpy import Scene
+    import glob
+
+    filenames = glob.glob('/test_data/W_XX-EUMETSAT-Darmstadt,SOUNDING+SATELLITE,METOPA+IASI_C_EUMC_20200204091455_68984_eps_o_so2_l2.bin')
+    scn = Scene(filenames=filenames, reader='iasi_l2_so2_bufr')
+    scn.load(['so2_height_3', 'so2_height_4'])
+    print(scn['so2_height_3'])
+
+
+Output:
+
+.. code-block:: none
+
+    <xarray.DataArray 'so2_height_3' (y: 23, x: 120)>
+    dask.array<where, shape=(23, 120), dtype=float64, chunksize=(1, 120), chunktype=numpy.ndarray>
+    Coordinates:
+        crs      object +proj=latlong +datum=WGS84 +ellps=WGS84 +type=crs
+    Dimensions without coordinates: y, x
+    Attributes:
+        sensor:               IASI
+        units:                dobson
+        file_type:            iasi_l2_so2_bufr
+        wavelength:           None
+        modifiers:            ()
+        platform_name:        METOP-2
+        resolution:           12000
+        fill_value:           -1e+100
+        level:                None
+        polarization:         None
+        coordinates:          ('longitude', 'latitude')
+        calibration:          None
+        key:                  #3#sulphurDioxide
+        name:                 so2_height_3
+        start_time:           2020-02-04 09:14:55
+        end_time:             2020-02-04 09:17:51
+        area:                 Shape: (23, 120)\nLons: <xarray.DataArray 'longitud...
+        ancillary_variables:  []
+
+References:
+
+Algorithm Theoretical Basis Document: https://acsaf.org/docs/atbd/Algorithm_Theoretical_Basis_Document_IASI_SO2_Jul_2016.pdf
+
+"""
+
+
+# TDB: this reader is based on iasi_l2.py and seviri_l2_bufr.py
+
+import logging
+from datetime import datetime
+import numpy as np
+import xarray as xr
+import dask.array as da
+
+try:
+    import eccodes as ec
+except ImportError:
+    raise ImportError(
+        """Missing eccodes-python and/or eccodes C-library installation. Use conda to install eccodes""")
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy import CHUNK_SIZE
+
+logger = logging.getLogger('IASIL2SO2BUFR')
+
+data_center_dict = {3: {'name': 'METOP-1'}, 4:  {'name': 'METOP-2'},
+                    5: {'name': 'METOP-3'}}
+
+
+class IASIL2SO2BUFR(BaseFileHandler):
+    """File handler for the IASI L2 SO2 BUFR product."""
+
+    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+        """Initialise the file handler for the IASI L2 SO2 BUFR data."""
+
+        super(IASIL2SO2BUFR, self).__init__(filename, filename_info, filetype_info)
+
+        start_time, end_time = self.get_start_end_date()
+
+        sc_id = self.get_attribute('satelliteIdentifier')
+
+        self.properties = {}
+        self.properties['start_time'] = start_time
+        self.properties['end_time'] = end_time
+        self.properties['SpacecraftName'] = data_center_dict[sc_id]['name']
```

Comment:
For consistency: we usually call this `metadata` or `mda` for short, not `properties.